### DISCUSSION TITLE
standards_and_practices: input-files data and referencing, traceability

### DISCUSSION THREAD

**Discussion by amargeride** [terrapower/armi#622] [General]
I wonder if you should add some recommendations about the input files format in this developer documentation:

https://github.com/terrapower/armi/blob/master/doc/developer/standards_and_practices.rst

Using .json files can be very practical but, since you cannot write comments, it can be difficult to track the data implemented in the file and make review and analyses preparation difficult.

json files are very efficient for a developer perspective but, for a user perspective, it could become cumbersome in term of quality if you cannot track each items by short comments like this:

```python
od: 3.14 # from drawing XXXX
keff_convergence: 0.00001 #from DIF3D manual YYYY
thickness_tolerance: 0.1 #od tol - id tol - porosity
```

To prepare a design analyses with safety requirements (and so quality), user would have to report the data origin (from a drawing, a specification, another analysis, a guideline, manual). With the usage of json file, user cannot specified the data origin through a simple comment which is very useful and efficient and simple to provide data origin or data explanation tips with some simple formula...

Therefore, I would recommend to add as warning or recommendations to use YAML file for user interface and keep json file for other input file that might not require traceability (it it exists). The recommendation could also be to suggest adding a field in the json file dedicated to the data origin report.

**Comment by john-science** [comment]
I am not sure the docs are totally correct. 

Right now we support YAML and XML, but I am in the process of removing XML.

We certainly don't support JSON settings or blueprints currently.  I think the goal is to move to YAML-only in the coming weeks.

**Comment by keckler** [comment]
FYI, the link in the original post is not public. Here is the public version of the page:
https://github.com/terrapower/armi/blob/master/doc/developer/standards_and_practices.rst

**Reply by john-science** [reply]
Fixed.  Thanks!
